src/huggingface_hub/repository.py

```diff
@@ -404,6 +408,10 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
         If this folder is a git repository with linked history, will try to update the repository.
         """
         token = use_auth_token if use_auth_token is not None else self.huggingface_token
+        if token is None and self.private:
+            raise ValueError(
+                "You need to provide a Hugging Face Token in `use_auth_token` to work with private repositories"
```

Comment:
nit: I think providing the token is more of an edge case for users working in colab or certain scenarios. Maybe we should mention that users should first login with `huggingface-cli login`  or manually provide a token.